lib/mongo/bulk_write.rb

```diff
@@ -182,11 +186,18 @@ def insert_one(documents, server, operation_id)
     end
 
     def update(documents, server, operation_id)
```

Comment:
Yes, I'm not quite done with this PR; it's a WIP. 
I've just commit BulkWrite delete support: https://github.com/mongodb/mongo-ruby-driver/pull/788/commits/6c94dbbe8e1b0fd91034f45967ad7aa6b407c97b